test/functional/unified-spec-runner/match.ts

```diff
@@ -235,32 +248,70 @@ export function specialCheck(
   }
 }
 
+// CMAP events where the payload does not matter.
+const EMPTY_CMAP_EVENTS = {
+  poolCreatedEvent: ConnectionPoolCreatedEvent,
+  poolClosedEvent: ConnectionPoolClosedEvent,
+  connectionCreatedEvent: ConnectionCreatedEvent,
+  connectionReadyEvent: ConnectionReadyEvent,
+  connectionCheckOutStartedEvent: ConnectionCheckOutStartedEvent,
+  connectionCheckOutFailedEvent: ConnectionCheckOutFailedEvent,
+  connectionCheckedOutEvent: ConnectionCheckedOutEvent,
+  connectionCheckedInEvent: ConnectionCheckedInEvent
+};
+
+function validEmptyCmapEvent(
+  expected: ExpectedCommandEvent | ExpectedCmapEvent,
+  actual: CommandEvent | CmapEvent
+) {
+  return Object.keys(EMPTY_CMAP_EVENTS).some(key => {
+    const eventType = EMPTY_CMAP_EVENTS[key];
+    return actual instanceof eventType;
+  });
+}
+
 export function matchesEvents(
-  expected: ExpectedEvent[],
-  actual: CommandEvent[],
+  expected: (ExpectedCommandEvent & ExpectedCmapEvent)[],
+  actual: (CommandEvent & CmapEvent)[],
   entities: EntitiesMap
 ): void {
-  // TODO: NodeJS Driver has extra events
-  // expect(actual).to.have.lengthOf(expected.length);
+  if (actual.length !== expected.length) {
+    // const actualNames = actual.map(a => a.constructor.name);
```

Comment:
this seems to be an empty code block

Reply:
Relevant convo over on the bigger PR: https://github.com/mongodb/node-mongodb-native/pull/2742#discussion_r668296050

Reply:
@nbbeeken Thanks! still think that we should maybe remove the if block and the code that attempts to do things and add a comment in the code summarizing this: 

> It was intentional as the Node driver does not execute a killCursors after a getMore has errored, which causes the absence of the killCursors events in some cases. I could not find anywhere so far in the specs that states an errored getMore requires an immediate killCursors afterwards, and all the driver does in this case is cleanup the cursor (but not explicitly close() it).

(maybe even file a ticket to explore this further, since if we know exactly what's out of the ordinary, we may be able to filter explicitly those cases out)

Reply:
This check is now back and asserting the event lengths, so no comment needed.

Reply:
It looks like some of the new snapshot tests are failing now with that check, I don't think you need to take care of it here because I need to sort out those tests anyway, but if you could add them to the skip list, that would allow the pipeline to stay green on main

Reply:
snapshot-sessions-unsupported-ops
- Server returns an error on findOneAndUpdate with snapshot
- Server returns an error on deleteOne with snapshot
- Server returns an error on updateOne with snapshot

Reply:
I've skipped the one failing test now.